src/apps/pulse/components/Buy/PreviewBuy.tsx

```
@@ -195,63 +312,223 @@ export default function PreviewBuy(props: PreviewBuyProps) {
     }
   };
 
+  // Relay Buy: execute buy directly
+  const executeBuyDirectly = async () => {
+    if (!buyToken || !buyOffer || !kit || !fromChainId) {
+      return;
```

Does this result in a silent failure?